src/operations/query/filters/canonical.js

```diff
@@ -22,14 +22,17 @@ function filterByCanonical({propertyObj, queryParameterValue, columns}) {
                 ),
             },
         );
+        // Adding propertyobj fields to the columns set, to be used as index hints
+        propertyObj.fields.forEach(columns.add, columns);
     } else {
         and_segments.push(
             {
                 [`${propertyObj.field}`]: queryParameterValue,
             }
         );
+        // Adding the field to columns set, to be used as index hints
+        columns.add(`${propertyObj.field}`);
     }
-    columns.add(propertyObj.fields ? `${propertyObj.fields}` : `${propertyObj.field}`);
     return and_segments;
```

Comment:
Instead of repeating the columns add logic in every if else block, we should write this logic at one place in the end of the function so that new if else blocks need not repeat the same logic. This comment applies to other files also.